javalib/src/main/scala/java/lang/Integer.scala

```diff
@@ -49,40 +62,79 @@ object Integer {
-    parseIntImpl(s, radix, signed = false)
-
+  private final val decimalScale: Array[scala.Int] = Array(1000000000,
+                                                           100000000,
+                                                           10000000,
+                                                           1000000,
+                                                           100000,
+                                                           10000,
+                                                           1000,
+                                                           100,
+                                                           10,
+                                                           1)
+  private final val digits = Array('0',
+                                   '1',
+                                   '2',
+                                   '3',
+                                   '4',
+                                   '5',
+                                   '6',
+                                   '7',
+                                   '8',
+                                   '9',
+                                   'a',
+                                   'b',
+                                   'c',
+                                   'd',
+                                   'e',
+                                   'f',
+                                   'g',
+                                   'h',
+                                   'i',
+                                   'j',
+                                   'k',
+                                   'l',
+                                   'm',
+                                   'n',
+                                   'o',
+                                   'p',
+                                   'q',
+                                   'r',
+                                   's',
+                                   't',
+                                   'u',
+                                   'v',
+                                   'w',
+                                   'x',
+                                   'y',
+                                   'z')
   @inline
-  private def parseIntImpl(
-      s: String, radix: scala.Int, signed: scala.Boolean): scala.Int = ???
-
-  @inline def toString(i: scala.Int): String =
-    ???
-
-  @inline def toUnsignedString(i: Int, radix: Int): String =
-    toStringBase(i, radix)
-
-  @inline def compare(x: scala.Int, y: scala.Int): scala.Int =
-    if (x == y) 0 else if (x < y) -1 else 1
+  private def parse(s: _String,
+                    offset: scala.Int,
+                    radix: scala.Int,
+                    negative: scala.Boolean): scala.Int = {
+    val max    = MIN_VALUE / radix
+    val length = s.length()
+    var result = 0
+    var offset = 1
+    while (offset < length) {
+      offset += 1
+      val digit = Character.digit(s.charAt(offset), radix)
```

Comment:
`offset++` first returns old value and then increments. It means that `offset += 1` should be after `val digit = ...`.